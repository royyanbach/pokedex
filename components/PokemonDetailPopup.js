import React, { Component } from "react";

import PokemonStatItem from './PokemonStatItem';

export default class PokemonDetailPopup extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="pokemon-detail-overlay">
        <div className="pokemon-detail-popup">
          <h2 className="pokemon-detail-popup__name">{ this.props.pokemonDetail.species.name }</h2>
          <p className="pokemon-detail-popup__types">{ this.props.pokemonDetail.types.map(item => item.type.name).join(', ') }</p>
          <img className="pokemon-detail-popup__image" src={ this.props.pokemonDetail.sprites.front_default } />
          <img className="pokemon-detail-popup__image" src={ this.props.pokemonDetail.sprites.back_default } />
          {
            this.props.pokemonDetail.stats.map((statsDetail, idx) => <PokemonStatItem key={ idx } statsData={ statsDetail }/>)
          }
          <button className="pokemon-detail-popup__close" onClick={ this.props.onClosePopup }>Close</button>
        </div>
      </div>
    )
  }
}
