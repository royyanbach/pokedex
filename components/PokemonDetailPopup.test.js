import React from 'react';
import renderer from 'react-test-renderer';

import PokemonDetailPopup from './PokemonDetailPopup';

it('Render pokemon popup detail correctly', () => {
  const pokemonDetail = {
    species: {
      name: 'bulbasaur',
    },
    types: [
      {
        type: {
          name: 'grass'
        }
      },
      {
        type: {
          name: 'poison'
        }
      },
    ],
    sprites: {
      front_default: '',
      back_default: '',
    },
    stats: [{
      base_stat: 40,
      stat: {
        name: 'hp',
      },
    }],
  };

  const component = renderer.create(
    <PokemonDetailPopup pokemonDetail={ pokemonDetail }/>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
