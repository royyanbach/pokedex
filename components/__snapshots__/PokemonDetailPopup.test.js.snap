// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render pokemon popup detail correctly 1`] = `
<div
  className="pokemon-detail-overlay"
>
  <div
    className="pokemon-detail-popup"
  >
    <h2
      className="pokemon-detail-popup__name"
    >
      bulbasaur
    </h2>
    <p
      className="pokemon-detail-popup__types"
    >
      grass, poison
    </p>
    <img
      className="pokemon-detail-popup__image"
      src=""
    />
    <img
      className="pokemon-detail-popup__image"
      src=""
    />
    <div
      className="pokemon-stat-item"
    >
      <p
        className="pokemon-stat-name"
      >
        hp
      </p>
      <p
        className="pokemon-stat-value"
      >
        <strong>
          40
        </strong>
        /100
      </p>
    </div>
    <button
      className="pokemon-detail-popup__close"
    >
      Close
    </button>
  </div>
</div>
`;
